client/landing/stepper/declarative-flow/plugin-bundle-flow.ts

```diff
@@ -129,11 +130,16 @@ const pluginBundleFlow: Flow = {
 		function submit( providedDependencies: ProvidedDependencies = {}, ...params: string[] ) {
 			recordSubmitStep( providedDependencies, intent, flowName, currentStep );
 
+			const defaultExitDest = siteDetails?.options
```

Comment:
It may be worth checking for the full property path to make sure we have a theme slug.
```suggestion
			const defaultExitDest = siteDetails?.options?.theme_slug
```